### PR TITLE
ENH: linalg: Add lwork wrapper for ?geqrf

### PR DESCRIPTION
**Pull request by Kai-Striega**
This commit adds the missing wrapper for ?geqrf_lwork as discussed in #11169. 

@ilayn would you please take a look over this? The main issue I came across is trying to find the required values to pass. Looking at ``ilaenv`` it requires each dimension `m`, `n` but is there a more general way to find the required arguments?

### CODE REVIEW

**Comment by ilayn** [comment]
Looks good to me!

> The main issue I came across is trying to find the required values to pass. Looking at ilaenv it requires each dimension m, n but is there a more general way to find the required arguments?

Not that I know of. LWORK query is specific to our f2py machinery hence I don't think LAPACK provides anything like that.

File: scipy/linalg/tests/test_lapack.py [hunk 1]
```diff
@@ -406,6 +406,19 @@ def test_gelsy(self):
                             rtol=25*np.finfo(dtype).eps)
 
 
+@pytest.mark.parametrize('dtype', DTYPES)
+def test_geqrf_lwork(dtype):
+    geqrf, geqrf_lwork = get_lapack_funcs(
+        ('geqrf', 'geqrf_lwork'),
+        dtype=dtype,
+    )
+    m, n = 4, 3
```
Comment (ilayn) [inline scipy/linalg/tests/test_lapack.py]: This is a quite fast query hence don't be shy to test with big numbers (less than 2**16 and maybe just to see what happens larger than 2**16).

Comment (Kai-Striega) [reply scipy/linalg/tests/test_lapack.py]: I gave it a run and it's failing for the complex128 with a memory error. The error comes from the call to ``np.ones`` - so it's not necessarily that the wrapper is failing  - but my hardware's ability to store the larger array.

Comment (ilayn) [reply scipy/linalg/tests/test_lapack.py]: You don't need to call the actual `geqrf`, we are calling it anyways with this wrapper. As long as info=0 or nonzero we are fine.

The time is spent in creating the ones matrix actually.

Comment (Kai-Striega) [reply scipy/linalg/tests/test_lapack.py]: Ok. I called ``geqrf(...,-1)`` first to have a value to check if it is correct. If I understand you correctly, we don't need to do that. I've got something on tonight - I'll push the changes once I get home.

File: scipy/linalg/tests/test_lapack.py [hunk 2]
```diff
@@ -406,6 +406,19 @@ def test_gelsy(self):
...
+    _, _, work, _ = geqrf(np.ones((m, n), dtype=dtype), lwork=-1)
+    lwork, info = geqrf_lwork(m=m, n=n)
+    assert_equal(info, 0)
+    assert_equal(lwork, work[0])
```
Comment (ilayn) [inline scipy/linalg/tests/test_lapack.py]: As this is failing on Travis, you might want to `int` them as they return in the dtype of the work. In production we use `_compute_lwork` function to strip off the integer out from the work query. Then you can call `assert_equal`.

**Comment by Kai-Striega** [comment]
@ilayn would you like any further changes or are you happy to merge?

File: scipy/linalg/flapack_gen.pyf.src [hunk 3]
```diff
@@ -978,6 +978,24 @@ subroutine <prefix>geqrf(m,n,a,tau,work,lwork,info)
 
 end subroutine <prefix>geqrf
 
+subroutine <prefix>geqrf_lwork(m,n,a,tau,work,lwork,info)
+    ! work, info = geqrf_lwork(m, n)
+    ! Calcualte the size of the ?geqrf work array.
```
Comment (mdhaber) [inline scipy/linalg/flapack_gen.pyf.src]:
```suggestion
    ! Calculate the optimal size of the ?geqrf work array.
```

Comment (Kai-Striega) [reply scipy/linalg/flapack_gen.pyf.src]: I'm not quite sure how that managed to slip through spell check =/ It's fixed now.

**Comment by ilayn** [comment]
Thanks all!

**Comment by mdhaber** [comment]
Thanks @Kai-Striega. Could you add this contribution to the 1.5.0 [release notes](https://github.com/scipy/scipy/wiki/*Release-note-entries-for-SciPy-1.5.0*)?

**Comment by mdhaber** [comment]
@Kai-Striega do you think this should go in the release notes?

**Comment by Kai-Striega** [comment]
@mdhaber yes I think it should. I can add an entry tonight.

**Pull request merged**